core/lib/zksync_core/src/eth_sender/eth_tx_aggregator.rs

```diff
@@ -403,50 +387,67 @@ impl EthTxAggregator {
     fn encode_aggregated_op(
         &self,
         op: &AggregatedOperation,
-        contracts_are_pre_boojum: bool,
+        contracts_are_pre_shared_bridge: bool,
     ) -> Vec<u8> {
-        let operation_is_pre_boojum = op.protocol_version().is_pre_boojum();
+        let operation_is_pre_shared_bridge = op.protocol_version().is_pre_shared_bridge();
+        assert_eq!(
+            contracts_are_pre_shared_bridge,
+            operation_is_pre_shared_bridge
+        );
+
+        let mut args = vec![Token::Uint(self.rollup_chain_id.as_u64().into())];
 
-        // For "commit" and "prove" operations it's necessary that the contracts are of the same version as L1 batches are.
-        // For "execute" it's not required, i.e. we can "execute" pre-boojum batches with post-boojum contracts.
         match op.clone() {
             AggregatedOperation::Commit(op) => {
-                assert_eq!(contracts_are_pre_boojum, operation_is_pre_boojum);
-                let f = if contracts_are_pre_boojum {
-                    &self.functions.pre_boojum_commit
-                } else {
+                if contracts_are_pre_shared_bridge {
                     self.functions
-                        .post_boojum_commit
+                        .pre_shared_bridge_commit
+                        .encode_input(&op.into_tokens())
+                        .expect("Failed to encode commit transaction data")
+                } else {
+                    let func = self
+                        .functions
+                        .post_shared_bridge_commit
                         .as_ref()
```

Comment:
I'm surprised that this is differnt than the pre_shared_bridge flow (where we call 'encode_input' directly. Any reason for that?

Reply:
The post shared bridge functions require an extra argument - the chain id.